pkg/api/upload.go

```diff
@@ -110,14 +113,46 @@ func (x UploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if r.Header.Get("Content-Type") != "application/json" {
-		invalidContent(w)
-		return
+	var body bytes.Buffer
+	switch r.Header.Get("Content-Encoding") {
+	case "application/gzip":
+		gzipReader, err := gzip.NewReader(r.Body)
+		if err != nil {
+			logger.WithError(err).Error("gzip.NewReader() failed")
+			badRequest(w, "")
+		}
+		if _, err := body.ReadFrom(gzipReader); err != nil {
+			logger.WithError(err).Error("gzipReader.Read() failed")
+			badRequest(w, "")
+		}
+		if err := gzipReader.Close(); err != nil {
+			logger.WithError(err).Error("gzipReader.Close() failed")
+			badRequest(w, "")
+		}
+
+	default:
+		if _, err := body.ReadFrom(r.Body); err != nil {
+			logger.WithError(err).Error("gzip.Read() failed")
+			badRequest(w, "")
+		}
 	}
 
 	var documents []Upload
-	if ok := jsonDecode(r.Body, &documents); !ok {
-		badRequest(w, "")
+	switch r.Header.Get("Content-Type") {
+	case "application/octet-stream":
+		if ok := gobDecode(&body, &documents); !ok {
+			badRequest(w, "")
+			return
+		}
```

Comment:
Maybe we should use a custom MIME type

Reply:
Do you have one you recommend?

Reply:
Maybe `application/x.vvgo.gob-upload` or `application/x.vvgo.upload`.

Ref: https://tools.ietf.org/html/rfc6838#section-3.4

Reply:
:ok: Lets use that!

Reply:
:thinking: Actually, lets us the golang pkg namespacing for this? Idk... I guess it doesn't really matter too much right now.